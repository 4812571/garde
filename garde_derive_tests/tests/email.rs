#[path = "./util/mod.rs"]
mod util;

#[derive(Debug, garde::Validate)]
struct Test<'a> {
    #[garde(email)]
    field: &'a str,
}

#[test]
fn email_valid() {
    util::check_ok(
        &[Test {
            field: "email@here.com",
        }],
        &(),
    )
}

#[test]
fn email_invalid() {
    util::check_fail!(
        &[Test {
            field: "invalid.com"
        }],
        &()
    )
}
