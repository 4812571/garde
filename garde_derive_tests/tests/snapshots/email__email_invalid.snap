---
source: garde_derive_tests/tests/email.rs
expression: snapshot
---
Test {
    field: "invalid.com",
}
`field` is not a valid email: value is missing `@`


